syntax = "proto3";
package pt.ulisboa.tecnico.nameserver.contract;

message RegisterRequest {
    string service = 1;
    string qualifier = 2;
    string address = 3;
}

message RegisterResponse {
    string response = 1;
}

message LookupRequest {
    string service = 1;
    string qualifier = 2;
}

message LookupResponse {
    repeated string address = 1;
}

message DeleteRequest {
    string service = 1;
    string address = 2;
}

message DeleteResponse {
    string response = 1;
}

service NameServer {
    rpc register(RegisterRequest) returns (RegisterResponse);
    rpc lookup(LookupRequest) returns (LookupResponse);
    rpc delete(DeleteRequest) returns (DeleteResponse);
}
